test/com/facebook/buck/swift/testdata/swift_test_with_host/.buckconfig

```diff
@@ -0,0 +1,5 @@
+[cxx]
+default_platform = iphonesimulator-x86_64
```

Comment:
We definitely want a separate test on macOS where we should force static linking of the Swift runtime (I think we can do that by setting `link_style = "static",` on the `apple_library` that contains Swift).

Reply:
When looking at this, it seems like buck currently doesn't support osx tests with a `bundle_loader`/`test_host_app`. 

When I make a test for this, everything builds but when running it runs something like `fbxctest -sdk macosx run-tests -appTest /path/to/test.xctest:/path/to/app.app`, which fails fbxctest with `UITests are not supported on OS X`. I tried updating `fbxctest` in buck using the script at `scripts/update-xctest.sh` to take in facebook/FBSimulatorControl@67bb15bd0d5eb58730b045fc6892902c2671dac0, but afterwards it tries to run as a uitest.

This leaves the question (I'm less familiar with tests on osx), does osx support the application tests in which you compile a test bundle with a `bundle_loader`? If not than this blacklisting code does not apply, and we should probably throw an error when trying to build a test a bundle with `test_host_app` for a macosx target.

Reply:
@robbertvanginkel I remember helping @marekcirkos with his efforts to add UI tests for macOS last year but my memory is a bit fuzzy. I know that we currently support logic tests without `test_host_app` and UI tests where both `test_host_app` and `ui_test_target_app` are specified (only on macOS 10.13+).

As long as we don't break logic tests with Swift on macOS, we should be good to go.

Reply:
Looked at this again, now it has a test on macos that validates the swift stdlib was linked statically.